test/shared/Deploy.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import {Vm} from "forge-std/Vm.sol";
+import {IPositionDescriptor} from "../../src/interfaces/IPositionDescriptor.sol";
+import {IPositionManager} from "../../src/interfaces/IPositionManager.sol";
+import {StateView} from "../../src/lens/StateView.sol";
+import {IV4Quoter} from "../../src/interfaces/IV4Quoter.sol";
+
+library Deploy {
+    Vm internal constant vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    function positionManager(
+        address poolManager,
+        address permit2,
+        uint256 unsubscribeGasLimit,
+        address positionDescriptor_,
+        address wrappedNative,
+        bytes memory salt
+    ) internal returns (IPositionManager manager) {
+        bytes memory args = abi.encode(poolManager, permit2, unsubscribeGasLimit, positionDescriptor_, wrappedNative);
+        bytes memory initcode = abi.encodePacked(vm.getCode("PositionManager.sol:PositionManager"), args);
+        assembly {
+            manager := create2(0, add(initcode, 0x20), mload(initcode), salt)
+        }
+    }
+
+    // no interface available, wrap in contract, this results in both versions being compiled, that's why we are importing the default version explicitly from the json file
+    function stateView(address poolManager) internal returns (StateView stateView_) {
+        bytes memory args = abi.encode(poolManager);
+        bytes memory initcode = abi.encodePacked(vm.getCode("foundry-out/StateView.sol/StateView.default.json"), args);
+        assembly {
+            stateView_ := create(0, add(initcode, 0x20), mload(initcode))
```

Comment:
why aren't these ones using create2 as well?

Reply:
all contracts that were instantiated using `new Contract()` I used `create` for the deployer, if they were instantiated  with a salt though, I used create2

Reply:
cool, i changed it to use create2 for consistency